app/services/sign_in/attribute_validator.rb

```diff
@@ -78,25 +78,39 @@ def add_mpi_user
     def update_mpi_correlation_record
       return if auto_uplevel
 
+      user_attribute_mismatch_checks
+      update_profile_response = mpi_service.update_profile(user_identity_from_attributes)
+      unless update_profile_response&.ok?
+        handle_error('User MPI record cannot be updated', Constants::ErrorCode::GENERIC_EXTERNAL_ISSUE)
+      end
+    end
+
+    def user_attribute_mismatch_checks
       user_identity_from_attributes.icn ||= mpi_response_profile.icn
       attribute_mismatch_check(:first_name, first_name, mpi_response_profile.given_names.first)
       attribute_mismatch_check(:last_name, last_name, mpi_response_profile.family_name)
       attribute_mismatch_check(:birth_date, birth_date, mpi_response_profile.birth_date)
       attribute_mismatch_check(:ssn, ssn, mpi_response_profile.ssn, prevent_auth: true)
-      update_profile_response = mpi_service.update_profile(user_identity_from_attributes)
-      unless update_profile_response&.ok?
-        handle_error('User MPI record cannot be updated', Constants::ErrorCode::GENERIC_EXTERNAL_ISSUE)
-      end
     end
 
     def attribute_mismatch_check(type, credential_attribute, mpi_attribute, prevent_auth: false)
       return unless mpi_attribute
 
       if scrub_attribute(credential_attribute) != scrub_attribute(mpi_attribute)
         error = prevent_auth ? Errors::AttributeMismatchError : nil
+        csp_type = user_identity_from_attributes.sign_in[:service_name]
+        csp_uuid = if csp_type == 'logingov'
+                     user_identity_from_attributes.logingov_uuid
+                   else
+                     user_identity_from_attributes.idme_uuid
+                   end
+        error_context = { icn: user_identity_from_attributes.icn,
+                          type: csp_type,
+                          csp_uuid: csp_uuid }
```

Comment:
From the PR I just merged in, you should be able to use the function `credential_uuid` for this now